### DISCUSSION TITLE
How to specify required_resource_keys for ConfgiurableResource that depends on another resource.

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#18231] [Q&A] [answered]
Hi everyone!
I'm creating a Resource Class (*"_MySecondResource")_* that depends on another resource class ("_*MyFirstResource*_".

`class MySecondCass(ConfigurableResource):`

I didn't find a way to specify the _*required_resource_keys*_ for the ConfigurableResource,
How do you suggest to solve this resources chain?
I'd like to use the MyFirstResource that I have in the context.
Thanks :heart:

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C01U954MEER/p1700663865234969?thread_ts=1700663865.234969&cid=C01U954MEER).*

**Comment by benpankow** [comment, marked as answer]
You can encode this dependency with the following:

Provided MyFirstResource is also a ConfigurableResource, you just need to drop it in as an annotated field:

```python
class MySecondClass(ConfigurableResource):
    my_first_resource: MyFirstResource
```

Then, in your definitions:

```python
defs = Definitions(
    resources={
        "my_second_class": MySecondClass(my_first_resource=MyFirstResource(...))
    }
)
```

or

```python
my_first_resource = MyFirstResource(...)
defs = Definitions(
    resources={
        "my_first_resource": my_first_resource,
        "my_second_class": MySecondClass(my_first_resource=my_first_resource)
    }
)
```

with the latter setup, `my_first_resource` will show up in your list of resources
